version: '3.8'

services:
  taskger:
    container_name: taskger-server
    build: ./taskger
    volumes:
      - /app/node_modules
    env_file:
      - taskger/.env
    depends_on:
      - postgres
    ports:
      - "3002:3001"
    networks:
      - taskger-network
  postgres:
    container_name: taskger-db
    image: postgres:12.5-alpine
    environment:
      POSTGRES_USER: "postgres"
      POSTGRES_PASSWORD: "postgres"
    ports:
      - "5432:5432"
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U $POSTGRES_USER']
      timeout: 1s
      retries: 10
    volumes:
      - pgdata:/var/lib/postgresql/data
      - ./infra/docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d
    networks:
      - taskger-network
volumes:
  pgdata:
    driver: local
networks:
  taskger-network:
    driver: bridge
